From 5e317592b96c694b140214d11022430eeef0102e Mon Sep 17 00:00:00 2001
From: Saahil Tomar <quic_saahtoma@quicinc.com>
Date: Mon, 15 May 2023 14:14:11 +0530
Subject: [PATCH 212/281] mfd: qcom-spmi-pmic: Add remove API

Add remove API for unloading spmi-pmic module, as the SPMI
framework right now requires drivers under it to have a
remove API defined when removing them.

Change-Id: I83e560ce873e2208c485bb8dade24d0c56c59770
Signed-off-by: Saahil Tomar <quic_saahtoma@quicinc.com>
---
 drivers/mfd/qcom-spmi-pmic.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mfd/qcom-spmi-pmic.c b/drivers/mfd/qcom-spmi-pmic.c
index 7e2cd79d17eb..7ef9489b3a20 100644
--- a/drivers/mfd/qcom-spmi-pmic.c
+++ b/drivers/mfd/qcom-spmi-pmic.c
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-only
 /*
  * Copyright (c) 2014, The Linux Foundation. All rights reserved.
+ * Copyright (c) 2023 Qualcomm Innovation Center, Inc. All rights reserved.
  */
 
 #include <linux/device.h>
@@ -251,8 +252,11 @@ static int pmic_spmi_probe(struct spmi_device *sdev)
 
 MODULE_DEVICE_TABLE(of, pmic_spmi_id_table);
 
+static void pmic_spmi_remove(struct spmi_device *sdev) {}
+
 static struct spmi_driver pmic_spmi_driver = {
 	.probe = pmic_spmi_probe,
+	.remove = pmic_spmi_remove,
 	.driver = {
 		.name = "pmic-spmi",
 		.of_match_table = pmic_spmi_id_table,
-- 
2.17.1

